### PR TITLE
feat: add full feature parity

### PR DESCRIPTION
**Pull request by peterdeme**
- Full feature parity
- Tests

### CODE REVIEW

File: stream_chat/tests/async_chat/test_client.py [hunk 1]
```diff
@@ -563,11 +574,52 @@ async def test_update_blocklist(self, client: StreamChatAsync):
...
         assert response["status"] == "error"
         assert "invalid SQS url" in response["error"]
 
+    async def test_guest_user(self, client: StreamChatAsync, random_user: Dict):
+        try:
+            response = await client.set_guest_user({"user": {"id": str(uuid.uuid4())}})
+            assert "access_token" in response
+        except Exception:
+            # Guest user isn't turned on for every test app
+            pass
```
Comment (gumuz) [inline stream_chat/tests/async_chat/test_client.py:603]: this test is a bit useless like this, it will always pass

Comment (peterdeme) [reply stream_chat/tests/async_chat/test_client.py:603]: it is, but I don't know if I want to mess with app settings in tests

Comment (gumuz) [reply stream_chat/tests/async_chat/test_client.py:603]: even if the request succeeds but the assert fails, it will still be ignore, no?

Comment (peterdeme) [reply stream_chat/tests/async_chat/test_client.py:603]: that's actually a good point @gumuz 🤦 I'm making it catch `StreamException`

Comment (peterdeme) [reply stream_chat/tests/async_chat/test_client.py]: Pushed. ✅

**Review by gumuz — APPROVED** [review]

**Pull request merged**